x/denommetadata/proposal_handler.go

```diff
@@ -23,23 +23,27 @@ func NewDenomMetadataProposalHandler(k types.BankKeeper) govtypes.Handler {
 // HandleCreateDenomMetadataProposal is a handler for executing a passed denom metadata creation proposal
 func HandleCreateDenomMetadataProposal(ctx sdk.Context, k types.BankKeeper, p *types.CreateDenomMetadataProposal) error {
 
-	found := k.HasDenomMetaData(ctx, p.TokenMetadata.Base)
-	if found {
-		return types.ErrDenomAlreadyExists
-	}
+	for _, metadata := range p.TokenMetadata {
+		found := k.HasDenomMetaData(ctx, metadata.Base)
+		if found {
+			return types.ErrDenomAlreadyExists
```

Comment:
I would simply skip the already found and not stop the  others from registration (so just log it)